src/brevitas/core/function_wrapper/clamp.py

```diff
@@ -73,3 +81,203 @@ def __init__(self, min_val: float) -> None:
     @brevitas.jit.script_method
     def forward(self, x: Tensor):
         return x.clamp_min(self.min_val)
+
+
+class FloatClamp(brevitas.jit.ScriptModule):
+    """"
+    ScriptModule for clamping minifloat formats to their inf/NaN implementations.
+
+    Currently, inf/NaN codes have to be encoded through the mantissa.
+    I.e. setting inf to 1101.111 (E4M3) is not a valid code.
+    """
+
+    __constants__ = [
+        'exponent_bit_width',
+        'mantissa_bit_width',
+        'exponent_bias',
+        'nan_values',
+        'inf_values',
+        'saturating']
+
+    def __init__(
+            self,
+            exponent_bit_width: Tensor,
+            mantissa_bit_width: Tensor,
+            exponent_bias: Tensor,
+            nan_values: Optional[Tuple[str]] = None,
+            inf_values: Optional[Tuple[str]] = None,
+            saturating: bool = False) -> None:
+        super(FloatClamp, self).__init__()
+
+        self.exponent_bit_width = exponent_bit_width
+        self.mantissa_bit_width = mantissa_bit_width
+        self.exponent_bias = exponent_bias
+
+        self.nan_values = nan_values
+        self.inf_values = inf_values
+        self.saturating = saturating
+
+        # inf without NaN not possible
+        if self.inf_values is None and self.nan_values is None:
+            self.max_val_impl = StatelessBuffer(
+                max_float(
+                    self.exponent_bit_width(), self.mantissa_bit_width(), self.exponent_bias()))
+        elif self.inf_values is not None and self.nan_values is not None:
+            # we have values for NaN and inf, so initiate MaxValInfNaN
+            self.max_val_impl = MaxFloatInfNaN(
+                exponent_bit_width=self.exponent_bit_width,
+                mantissa_bit_width=self.mantissa_bit_width,
+                exponent_bias=self.exponent_bias,
+                nan_values=self.nan_values,
+                inf_values=self.inf_values,
+                saturating=self.saturating)
+        elif self.inf_values is None and self.nan_values is not None:
+            # we only have values for NaN, so initiate MaxValNaN
+            self.max_val_impl = MaxFloatNaN(
+                exponent_bit_width=self.exponent_bit_width,
+                mantissa_bit_width=self.mantissa_bit_width,
+                exponent_bias=self.exponent_bias,
+                nan_values=self.nan_values,
+                saturating=self.saturating)
+        else:
+            # no NaN values but inf values
+            raise RuntimeError('Minifloat Error: inf value cannot exist without NaN value.')
+
+        self.clamp_impl = CaseClamp(inf_values=self.inf_values, saturating=self.saturating)
+
+    @brevitas.jit.script_method
+    def forward(self, inp: Tensor):
+        # get max value for the minifloat config
+        max_value = self.max_val_impl()
+        return self.clamp_impl(inp, max_value)
+
+
+class MaxFloatInfNaN(brevitas.jit.ScriptModule):
```

Comment:
This class and MaxFloatNaN are identical except for:
`min_special_case = min(map(lambda x: int(x, 2), self.nan_values + self.inf_values))`

Seems a bit of a waste. Let's have a single class for both